File: entites/Seg.py
class Seg:
    def __init__(self, startingVertexNumber, endingVertexNumber, angle, lineDefNumber, directionSameAsLineDef, offset):
        self.startingVertexNumber = startingVertexNumber
        self.endingVertexNumber = endingVertexNumber
        self.angle = angle
        self.lineDefNumber = lineDefNumber
        self.directionSameAsLineDef = directionSameAsLineDef
        self.offset = offset

    def __str__(self):
        return f"{self.startingVertexNumber=}, {self.endingVertexNumber=}, {self.angle=}, {self.lineDefNumber=}, {self.directionSameAsLineDef=}, {self.offset=}"

    __repr__ = __str__

File: cpp_code/cpp_classes/Seg.cpp
#include "../headers/Seg.h"
Seg::Seg(uint16_t startingVertexNumber, uint16_t endingVertexNumber, uint16_t angle, uint16_t lineDefNumber, bool directionSameAsLineDef, uint16_t offset)
        : startingVertexNumber(startingVertexNumber), endingVertexNumber(endingVertexNumber), angle(angle), lineDefNumber(lineDefNumber), directionSameAsLineDef(directionSameAsLineDef), offset(offset)
    {}

Seg::Seg()
        : startingVertexNumber(0), endingVertexNumber(0), angle(0), lineDefNumber(0), directionSameAsLineDef(false), offset(0)
    {}

std::ostream& operator<<(std::ostream& os, const Seg& obj) {
        os << "Seg{ ";
        os << "startingVertexNumber: " << (int) obj.startingVertexNumber << " ";
        os << "endingVertexNumber: " << (int) obj.endingVertexNumber << " ";
        os << "angle: " << (int) obj.angle << " ";
        os << "lineDefNumber: " << (int) obj.lineDefNumber << " ";
        os << "directionSameAsLineDef: " << obj.directionSameAsLineDef << " ";
        os << "offset: " << (int) obj.offset << " ";
        os << "}";
        return os;
    }